blockchain/emulator.go

```diff
@@ -0,0 +1,252 @@
+package blockchain
+
+import (
+	"fmt"
+
+	"github.com/dapperlabs/flow-playground-api/model"
+	"github.com/onflow/cadence"
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/parser"
+	emu "github.com/onflow/flow-emulator"
+	"github.com/onflow/flow-emulator/storage/memstore"
+	"github.com/onflow/flow-emulator/types"
+	flowsdk "github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go-sdk/templates"
+	"github.com/pkg/errors"
+)
+
+// blockchain interface defines an abstract API for communication with the blockchain. It hides complexity from the
+// consumer and communicates using flow native types.
+type blockchain interface {
+	// executeTransaction builds and executes a transaction and uses provided authorizers for signing.
+	executeTransaction(
+		script string,
+		arguments []string,
+		authorizers []model.Address,
+	) (*types.TransactionResult, *flowsdk.Transaction, error)
+
+	// executeScript executes a provided script with the arguments.
+	executeScript(
+		script string,
+		arguments []string,
+	) (*types.ScriptResult, error)
+
+	// createAccount creates a new account and returns it along with transaction and result.
+	createAccount() (*flowsdk.Account, *flowsdk.Transaction, *types.TransactionResult, error)
+
+	// getAccount gets an account by the address and also returns its storage.
+	getAccount(address flowsdk.Address) (*flowsdk.Account, *emu.AccountStorage, error)
+
+	// deployContract deploys a contract on the provided address and returns transaction and result.
+	deployContract(address flowsdk.Address, script string) (*types.TransactionResult, *flowsdk.Transaction, error)
+}
+
+var _ blockchain = &emulator{}
```

Comment:
What's this for?

Reply:
yes that's an easy trick to make sure we can detect compliance with an interface during checking.